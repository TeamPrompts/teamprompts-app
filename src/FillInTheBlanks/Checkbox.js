import React from 'react';

function Checkbox({ id, mode, onChange, value }) {
  const inputId = `${id}-${value}`;
  return (
    <label htmlFor={inputId} className="cursor-pointer">
      <div className="relative">
        <input
          checked={mode === value}
          data-testid={inputId}
          id={inputId}
          name={id}
          onChange={onChange}
          type="checkbox"
          value={value}
          className="hidden"
        />

        <div className="toggle__line w-6 h-3 bg-gray-400 rounded-full shadow-inner" />

        <div className="toggle__dot absolute w-4 h-4 bg-gray-300 rounded-full shadow inset-y-0 left-0" />
      </div>
    </label>
  );
}

export default Checkbox;
