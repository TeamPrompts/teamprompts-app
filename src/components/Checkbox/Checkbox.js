import React from 'react';

export const values = {
  off: 'off',
  on: 'on'
};

// TODO: check the html
function Checkbox({ id, onChange, value }) {
  return (
    <label className="cursor-pointer" htmlFor={id}>
      <div className="relative">
        <input
          checked={value === values.on}
          className="hidden"
          data-testid={id}
          id={id}
          onChange={() => onChange(value)}
          type="checkbox"
        />
        <div className="toggle__line w-6 h-3 bg-gray-400 rounded-full shadow-inner" />
        <div className="toggle__dot absolute w-4 h-4 bg-gray-300 rounded-full shadow inset-y-0 left-0" />
      </div>
    </label>
  );
}

export default Checkbox;
