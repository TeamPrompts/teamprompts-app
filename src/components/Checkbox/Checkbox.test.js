import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';
import Checkbox, { values } from './Checkbox';

describe('Checkbox', () => {
  const ID = '0';
  const mockOnChange = jest.fn();
  const props = {
    id: ID,
    onChange: mockOnChange,
    value: values.on
  };

  it('to match snapshot', () => {
    const tree = create(<Checkbox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('click', () => {
    const { getByTestId } = render(<Checkbox {...props} />);

    fireEvent.click(getByTestId(ID));

    expect(mockOnChange).toBeCalledWith(values.on);
  });
});
