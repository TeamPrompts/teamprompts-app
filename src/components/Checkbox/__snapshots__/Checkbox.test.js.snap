// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox to match snapshot 1`] = `
<span
  className="inline-block ml-3 relative"
  onClick={[Function]}
>
  <input
    checked={true}
    className="hidden"
    data-testid="0"
    id="0"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="cursor-pointer"
    htmlFor="0"
  >
    <div
      className="w-6 h-3 bg-gray-400 rounded-full shadow-inner"
    />
    <div
      className="toggle__dot absolute w-4 h-4 bg-gray-300 rounded-full shadow inset-y-0 left-0 toggle__dot--checked"
    />
  </label>
</span>
`;
